src/go/rpk/pkg/config/config_test.go

```diff
@@ -289,6 +289,13 @@ redpanda:
   admin:
     address: 0.0.0.0
     port: 9644
+  archival_storage_enabled: false
+  archival_storage_max_connections: 0
```

Comment:
These don't look like good defaults. Most of the archival configs are optional, so I think you need to make them pointers with `omitempty`.

Reply:
Makes sense. I made them pointers and using `omitempty` now. Let me know if you have any other comments.